R/render-table.R

```diff
@@ -37,23 +108,96 @@ renderTable <- function(expr, ..., env=parent.frame(), quoted=FALSE, func=NULL)
     xtable_args <- dots[intersect(names(dots), xtable_argnames)]
     non_xtable_args <- dots[setdiff(names(dots), xtable_argnames)]
 
-    # Call xtable with its args
+    # A small helper function to determine if the row.names can be coerced
+    # to numeric or if they are legitimate strings
+    isNumber <- function(n) {
+      !(suppressWarnings(is.na(all(n == as.character(as.numeric(n))))))
+    }
+
+    # Figure out column alignment
+    ## Case 1: if align=NULL, check if row.names are numbers. If not, make
+    ## sure to left-align them (xtable right-aligns them by default, which
+    ## looks weird when the row.names are strings).
+    if (is.null(align)) {
+      cols <- if (isNumber(row.names(data))) "r" else "l"
+      for (i in seq_len(ncol(data))) {
+        cls <- class(data[,i])
+        if (cls == "numeric" || cls == "integer") cols <- paste0(cols, "r")
+        else cols <- paste0(cols, "l")
```

Comment:
This is a dangerous test for class because the class isn't always a length 1 vector. It's safer to use `inherits()`, or in this case I think you could just use `is.numeric()`.  I'd do something like:

```R
is_num <- vapply(data, is.numeric, logical(1))
cols <- ifelse(is_num, "r", "l")
```